tests/simple_sum.asm

```diff
@@ -0,0 +1,129 @@
+// Verfies that a sum in the input has been computed properly.
+// Input: sum, cnt, x_1, x_2, ..., x_cnt
+// This input is assumed to be present in a minirust variable called "input"
+// of type "Vec<FieldElement>"
+
+// Code in `${`...`}` is rust-like code that is run by the prover
+// to generate free inputs.
+
+reg A;
+reg CNT;
+reg pc[@pc]; // "@pc" means "pc' = pc + 1" is the default propagation (instead of pc' = pc) and it tracks the line in the program.
+reg X[<=]; // "<=" means it is the default assignment register.
+
+// Code in `pil{`..`}` is pil code that is inserted into the pil file.
+pil{
+    col witness XInv;
+    col XIsZero = 1 - X * XInv;
+    XIsZero * X = 0;
+}
+
+instr jmpz <=X= c, l: label { pc' = XIsZero * l + (1 - XIsZero) * (pc + 1) }
+instr jmp l: label { pc' = l }
+instr dec_CNT { CNT' = CNT - 1 }
+instr assert_zero <=X= a { XIsZero = 1 }
+
+CNT <=X= ${ input[1] };
+
+start::
```

Comment:
Double colon because otherwise we get a parsing ambiguity with `:public` being references to public variables. I don't think references should include the kind of reference, though. I'll think how public variable references could be parsed differently depending on the context.